app/resources/v1/pacbio/library_resource.rb

```diff
@@ -72,10 +72,12 @@ class LibraryResource < JSONAPI::Resource
       #   @return [Float] the used volume of the library
       # @!attribute [rw] available_volume
       #   @return [Float] the available volume of the library
+      # @!attribute [r] barcode
```

Comment:
we have this as r but it is not marked as ready only. Is this an issue?

Reply:
You will get a NoMethodError if you try and set this value as its being delegated to tube, but delegation does not allow you to set the values.

Reply:
Ill override the creatable and updatable fields so you can't use barcode, that way we will get a better error.

Reply:
Updated in [027bc54](https://github.com/sanger/traction-service/pull/1587/commits/027bc544038ef93648319046311cf8d9dfda4e98)